### DISCUSSION TITLE
Question about deselect option

### DISCUSSION THREAD

**Discussion by Renaud6322** [davidstutz/bootstrap-multiselect#1275] [General]
Hi and thanks for this nice plugin.
I would like to unselect all options if a specific one is selected.
Is it possible ?
I did that but it's not working (just tried to deselect an option for test)

```
onChange:function (option, checked) {
	if(option.val() == 'memo' && checked == true){
		console.log(option.val() + " " +checked);
		$('#select-dests').multiselect('deselect', 'client');

	}
},
```

Could you help me ?
Best regards.
Renaud.

**Comment by tiesont** [comment]
`deselect` un-selects the option matching the value that was provided as the second parameter. If you're trying to un-select every option, you want to use `deselectAll`:

> Deselects all options. If justVisible is set to true or not provided, all visible options are deselected, otherwise (justVisible set to false) all options are deselected.

**Comment by Renaud6322** [comment]
Thank you so much for the answer !

This code              

```javascript
onChange:function (option, checked) {
    if(option.val() == 'memo' && checked == true){
       $('#select-dests').multiselect('deselectAll');
    }
},
```

Uncheck all checkboxes when "memo" is checked, it's parefect.
But the problem is "memo" is uncheckeable.

**Reply by tiesont** [reply]
If "memo" is not an option which can be selected, how will `if(option.val() == 'memo' && checked == true)` ever be true?

As you may have noticed, I converted this to a Discussion post - this isn't an issue with the plugin.

**Reply by Renaud6322** [reply]
In fact, I just want to write "if the selected option, then disable all others". But I think it's not the right syntax.

**Reply by Renaud6322** [reply]
And, of course, this is not a bug of the plugin. Only a bug of my mind :-(

**Reply by tiesont** [reply]
Do you want to _disable_ the other options, or just uncheck them?

**Reply by Renaud6322** [reply]
I would like : if option "memo" is checked, the all others are unchecked and not checkeable. It's just impossible to select "memo" with any other option

**Comment by tiesont** [comment]
I suspect you're after something like this (untested):

```javascript
onChange:function (option, checked) {
    if(option.val() == 'memo' && checked == true){
        console.log(option.val() + " " + checked);
        
        $('#select-dests option:not([value="memo"])').prop({ selected: false, disabled: true });
        
        $('#select-dests').multiselect('refresh');
    }
    else {
        $('#select-dests option').prop({ disabled: false });
        
        $('#select-dests').multiselect('refresh');
    }
},
```

This will un-select and disable all options in this select which do not have the value "memo". Since we're manipulating the actual `<select>` element, we need the plugin to update itself to match, which is what the 'refresh' method does:

> This method is used to refresh the checked checkboxes based on the currently selected options within the select.

I added the `else` block to basically reverse this if "memo" becomes un-selected - not sure if that's a valid use-case for you.

Please note that none of this is tested, but I've done similar things in the past, so with some tweaks on your end it should work.

**Comment by Renaud6322** [comment]
Hi and many thanks, it's exactly what I wanted to do. Perfect !
